appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -1012,6 +1012,38 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
   };
   menuOptions.push(enableAll);
 
+  // Clear unused blocks
+  var clearUnusedBlocks = {enabled: true};
+  clearUnusedBlocks.text = Blockly.Msg.REMOVE_UNUSED_BLOCKS;
+  clearUnusedBlocks.callback = function() {
+    Blockly.Events.setGroup(true);
```

Comment:
The start of the event group should only happen when the user confirms the deletion on line 1036. You'll also want to wrap the changes in a try-finally block, e.g.:

```
try {
  Blockly.Events.setGroup(true);
  // manipulate the workspace
} finally {
  // clean up the group regardless of whether an error occurs
  Blockly.Events.setGroup(false);
}
```

This prevents the event group from being left open if for some reason deleting a block causes an exception.